src/main/java/uk/gov/pay/api/service/ConnectorUriGenerator.java

```diff
@@ -24,9 +27,9 @@ public ConnectorUriGenerator(PublicApiConfig configuration) {
         this.configuration = configuration;
     }
 
-    String chargesURI(Account account, String agreementId) {
+    String chargesURI(Account account, CreatePaymentRequest createCardPaymentRequest) {
```

Comment:
I think this could just take `TokenPaymentType`

Reply:
yes good idea.